homeassistant/components/trest_solar/config_flow.py

```diff
@@ -0,0 +1,70 @@
+"""Config flow for TrestSolarController integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from trest_identity import TrestIdentityService
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate if the user input contains valid credentials."""
+
+    identity = TrestIdentityService(hass, data[CONF_USERNAME], data[CONF_PASSWORD])
+    await identity.authenticate_async()
+
+    if identity.check_token_is_expired():
+        raise InvalidAuth
+
+    return {"title": "Trest Solar Controller"}
+
+
+class TrestSolarControllerConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TrestSolarController."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
How can I check if there are any added user with the same username already? :)

Reply:
Is there something truly unique we know about the user? Preferably an unique user identifier. Otherwise we'd have to use the username, and then we have to know if the username is changeable, if so, we have to do another approach

Reply:
The unique thing is the username, it is the email of the user, and that is always unique

Reply:
Is it changeable?

Reply:
Nope

Reply:
Then set that as the unique id of the config flow

Reply:
How do I set this to unique?